app/controllers/concerns/verify_setup_completed.rb

```diff
@@ -0,0 +1,45 @@
+module VerifySetupCompleted
+  extend ActiveSupport::Concern
+
+  MANDATORY_CONFIGS = %i[
+    community_description
+    community_action
+    tagline
+
+    main_social_image
+    logo_png
+
+    mascot_user_id
+    mascot_image_url
+
+    meta_keywords
+
+    suggested_tags
+    suggested_users
+  ].freeze
+
+  included do
+    # rubocop:disable Rails/LexicallyScopedActionFilter
+    before_action :verify_setup_completed, only: %i[index new edit show]
+    # rubocop:enable Rails/LexicallyScopedActionFilter
+  end
+
+  private
+
+  def verify_setup_completed
+    return if setup_completed? || config_path?
```

Comment:
Can we flip these? `setup_completed?` will make a call to Redis while `config_path?` doesn't make any external calls and if`config_path?` returns true we can skip the Redis call.

I really wish there was a way we could verify setup completed without having to hit Redis on every call 🤔 I'm not sure of another way though off the top of my head. Maybe something to keep in mind.

Reply:
Good point, I recently suggested a similar change elsewhere, so 👍 on catching that! 

I'm a bit confused about the mention of Redis though. `SiteConfig` is stored in the DB. It also caches its values for a while, so they aren't retrieved every time:

```ruby
[1] dev.to(main)> SiteConfig.public_send(:logo_png).present?
  SiteConfig Load (0.6ms)  SELECT var, value FROM "site_configs"
true
[2] dev.to(main)> SiteConfig.public_send(:secondary_logo_url).present?
false
```

I just woke up and haven't had enough coffee yet, so am probably missing something, can you elaborate which Redis calls you mean?

Reply:
@mstruve I switched to `return if config_path? || setup_completed?` now. It's probably not going to make a lot of difference since only `/internal/config` will trigger this early return, but it's a good habit to sort them like this anyway.

Reply:
@citizen428 SiteConfig caches things in memory and in Redis, it's a two layer cache system:

https://github.com/huacnlee/rails-settings-cached/blob/7ae49b0f46a14d7b0a065f0719650c168d4b1505/lib/rails-settings/base.rb#L130-L138

```ruby
          RequestStore.store[:rails_settings_all_settings] ||= begin
            Rails.cache.fetch(self.cache_key, expires_in: 1.week) do
              vars = unscoped.select("var, value")
              result = {}
              vars.each { |record| result[record.var] = record.value }
              result.with_indifferent_access
            end
          end
        end
```

Reply:
Thank you @rhymes 😃